src/GNDStk/defaulted/test/value.test.cpp

```diff
@@ -0,0 +1,75 @@
+
+#include "catch.hpp"
+#include "GNDStk.hpp"
+
+SCENARIO("Testing defaulted<T>'s value retrieval functions") {
+   using njoy::GNDStk::defaulted;
+
+   // ------------------------
+   // With
+   // defaulted(default)
+   // ------------------------
+
+   WHEN("We construct a defaulted(default value)") {
+      const defaulted<int> d(123);
+      const defaulted<std::vector<double>> v({ 1.2, 3.4, 5.6, 7.8 });
+
+      // value()
+      THEN("value() works as expected") {
+         CHECK(d.value() == 123);
+      }
+
+      // *
+      THEN("operator* works as expected") {
+         CHECK(*d == 123);
+      }
+
+      // ->
+      THEN("operator-> works as expected") {
+         CHECK(v->size() == 4);
+      }
+
+      // value()
+      THEN("conversion to T works as expected") {
+         CHECK(int(d) == 123);
```

Comment:
Ah, here is where I see how the operator is used. It isn't clear from the function signature this is how it should be used. Perhaps that's my C++ ignorance showing.

Reply:
You probably just haven't written too many C++ conversion operators. Google, say, "c++ user defined conversion operator" to get as much information as you want. It's one of those things that any given person might or might not have used much in their own codes.